apis/intermediate_api/intermediate_api/api/core_api.py

```diff
@@ -23,68 +23,126 @@
 
 # Site Selection
 @app.route(const.CORE_API_ENSEMBLE_IDS_ENDPOINT, methods=["GET"])
-@decorators.requires_auth
+@decorators.get_authentication
 def get_ensemble_ids(is_authenticated):
-    return utils.proxy_to_api(
-        request, const.ENSEMBLE_IDS_ENDPOINT, "GET", CORE_API_BASE, CORE_API_TOKEN,
-    )
+    if is_authenticated and auth0.requires_permission("hazard"):
+        return utils.proxy_to_api(
+            request, const.ENSEMBLE_IDS_ENDPOINT, "GET", CORE_API_BASE, CORE_API_TOKEN,
+        )
+    else:
+        raise auth0.AuthError(
```

Comment:
Can probably make this exception a constant? Also you technically don't need the else

Reply:
Man, we can even have them as a constant? will check that out.

> Also you technically don't need the else
Yeah as if it doesn't meet if statement condition, will hit this `raise` regardless but added purely for readability :joy: 
It looked slightly uncomfortable with a different indentation...